pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,427 @@
+                cvb.wheelEvent)
+            self._signalConnectionsByChart[chart.name]["tvb.sigHistoryChanged"] = tvb.sigHistoryChanged.connect(
+                cvb.scaleHistory)
+
+    def disconnect_all(self, chart):
+        """Disconnects all signals related to this widget for the given chart."""
+        for conn_name, conn in self._signalConnectionsByChart[chart.name].items():
+            if conn is not None:
+                QObject.disconnect(conn)
+                self._signalConnectionsByChart[chart.name][conn_name] = None
+
+    def clean(self):
+        """Clears all charts' contents."""
+        # CLEAR PLOTS
+        for p in self.charts.values():
+            p.clear()
+        self.update()
+
+    def getPlotItem(self, name=None):
+        """Get the PlotItem associated to the chart of given name.
+
+        Parameters
+        ----------
+        name : str or None
+            The name of the chart to select.
+            If None the default one will be selected.
+
+        Returns
+        -------
+        PlotItem
+            The PlotItem associated to the selected chart.
+        """
+        if name is None:
+            return self.pi
+        else:
+            return self.charts[name].plotItem
+
+    def setAxisRange(self, axis_name, range=None, **kwargs):
+        """Sets the range of the axis with given name.
+
+        Parameters
+        ----------
+        axis_name : str or None
+            The name of the axis to select.
+        range : list of int or float or None
+            The range to set to the axis to selected.
+            If None: autorange will be enabled.
+            If list of len 1: range will be set between 0 and range[0]
+            If list of len 2: range will be set between the two range values
+        """
+        if range is None or len(range) == 0:
+            # AUTORANGE
+            range = None
+        elif len(range) == 1:
+            # ZERO TO R
+            range = [min(0, *range), max(0, *range)]
+        elif len(range) == 2:
+            # SET GIVEN RANGE
+            range = [min(*range), max(*range)]
+        else:
+            raise AttributeError("bad range")
+        if range is None:
+            self.enableAxisAutoRange(axis_name)
+        else:
+            self.disableAxisAutoRange(axis_name)
+            axis = self.axes[axis_name]
+            charts = [self.charts[chart] for chart in axis.charts]
+            if axis.orientation in ["top", "bottom"]:  # IS X AXIS
+                for chart in charts:
+                    vb = chart.plotItem.vb
+                    vb.setXRange(*range, **kwargs)
+            elif axis.orientation in ["left", "right"]:  # IS Y AXIS
+                for chart in charts:
+                    vb = chart.plotItem.vb
+                    vb.setYRange(*range, **kwargs)
+
+    def update(self):
+        """Updates all charts' contents."""
+        for axis_name, axis in self.axes.items():
```

Comment:
given we don't use `axis_name` we can probably just iterate through `self.axes.values()` ?

Reply:
done